### PR TITLE
chore(deps): upgrade beam 2.28.0

### PR DESCRIPTION
**Pull request by igorbernstein2**
New version of beam requires a region now. So to avoid breaking our existing users I added code that will infer the region from the zone.
I also removed the docs for beam specific options to avoid these kinds of issues in the future

### CODE REVIEW

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 1]
```diff
@@ -140,21 +120,18 @@ limitations under the License.
...
-      <version>${beam-auto-value.version}</version>
+      <version>1.7.4</version>
       <scope>provided</scope>
     </dependency>
 
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <version>${gcs-guava.version}</version>
+      <version>30.1-jre</version>
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml:130]: to confirm, this a 1.8 module?

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/sequencefiles/Utils.java [hunk 2]
```diff
@@ -46,6 +47,21 @@ public static PipelineOptions tweakOptions(PipelineOptions opts) {
     }
     DataflowPipelineOptions dataflowOpts = opts.as(DataflowPipelineOptions.class);
 
+    // Region is a newly added requirement in newer versions of dataflow runner.
+    // Make the pipeline backwards compatible by inferring the region from the zone.
+    // This is done by chopping off the last dash
+    if (Strings.isNullOrEmpty(dataflowOpts.getRegion())) {
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/sequencefiles/Utils.java:53]: please add a test for this

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml (hunk 1 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-beam-import/pom.xml:130]: yep, we can use java8 in beam

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/sequencefiles/Utils.java (hunk 2 above)
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/sequencefiles/Utils.java:53]: added, it actually caught a typo :)

**Review by kolea2 — APPROVED** [review]

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 3]
```diff
@@ -140,21 +120,18 @@ limitations under the License.
     <dependency>
       <groupId>com.google.auto.value</groupId>
       <artifactId>auto-value</artifactId>
-      <version>${beam-auto-value.version}</version>
+      <version>1.7.4</version>
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-dataflow-parent/bigtable-beam-import/pom.xml:123]: nope, we beam now uses vendored versions of these, so we are free to upgrade all we want

**Pull request merged**